libcodechecker/util.py

```diff
@@ -172,6 +172,49 @@ def get_default_workspace():
     return workspace
 
 
+def __strip_protocol_from_url(url):
+    """
+    Strip the PROTOCOL specifier from an URL string and return it, along with
+    the remainder of the URL.
+    """
+    if '://' not in url:
+        return None, url
+
+    parts = url.split('://', 1)
+    protocol = parts[0]
+    url = url.replace(parts[0] + '://', '').lstrip('/').rstrip('/')
```

Comment:
Why do you need the `lstrip` and `rstrip`? Maybe you could just use `strip`? Also, why replace? Shouldn't we ensure that only the beginning of the url is considered? Shouldn't we use `urlparse` for parsing?

Reply:
`urlparse` is not good with optional protocol specifiers.

```py
> urlparse.urlparse("localhost:8001/Default")
ParseResult(scheme='localhost', netloc='', path='8001/Default', params='', query='', fragment='')
> urlparse.urlparse("xazax://localhost:8001/Default")
ParseResult(scheme='xazax', netloc='localhost:8001', path='/Default', params='', query='', fragment='')
```